compiler/qsc_frontend/src/typeck/tests.rs

```diff
@@ -4695,3 +4695,52 @@ fn field_access_not_ident() {
         "##]],
     );
 }
+
+#[test]
+fn stressing_generic_inference() {
```

Comment:
Should this be in the benches instead of a unit test? (Since the bug is really about perf -  if we regress this scenario badly, we may not notice it if it's just another unit test, but we'd notice the perf regression if it was in the benches)

Reply:
That's a good question... I suppose it makes more sense as a benchmark, but also we don't pay that close attention to the benchmarks these days. Still, seems worth moving it since it is a better fit there.

Reply:
Moved into type checking benchmark.